pkg/cloud/services/networking/network.go

```diff
@@ -38,6 +39,52 @@ func (c createOpts) ToNetworkCreateMap() (map[string]interface{}, error) {
 	return gophercloud.BuildRequestBody(c, "network")
 }
 
+func (s *Service) ReconcileExternalNetwork(openStackCluster *infrav1.OpenStackCluster) error {
+
+	if openStackCluster.Spec.ExternalNetworkID != "" {
+		externalNetwork, err := s.getNetworkByID(openStackCluster.Spec.ExternalNetworkID)
+		if err != nil {
+			return err
+		}
+		if externalNetwork.ID != "" {
+			openStackCluster.Status.ExternalNetwork = &infrav1.Network{
+				ID:   externalNetwork.ID,
+				Name: externalNetwork.Name,
+			}
+			return nil
+		}
+	}
+
+	// ExternalNetworkID is not given
+	iTrue := true
+	networkListOpts := networks.ListOpts{}
+	listOpts := external.ListOptsExt{
+		ListOptsBuilder: networkListOpts,
+		External:        &iTrue,
+	}
+
+	allPages, err := networks.List(s.client, listOpts).AllPages()
+	if err != nil {
+		return err
+	}
+	allNetworks, err := networks.ExtractNetworks(allPages)
+	if err != nil {
+		return err
+	}
+
+	switch len(allNetworks) {
+	case 0:
+		return fmt.Errorf("external network not found")
+	case 1:
+		openStackCluster.Status.ExternalNetwork = &infrav1.Network{
+			ID:   allNetworks[0].ID,
+			Name: allNetworks[0].Name,
+		}
+		return nil
+	}
```

Comment:
should we at least indicate we select the network in log?

Reply:
Thanks, updated.